### ISSUE TITLE
mpi4py: Regression in spawn tests

### ISSUE THREAD

**Issue by dalcinl** [open-mpi/ompi#11378]
There is a new mpi4py regression with current **main** branch [[link]](https://github.com/mpi4py/mpi4py-testing/actions/runs/4089040622/jobs/7053982157#step:18:1409), the automated build that ran last week was OK. 

Output from one of the failing tests:
```
testArgsOnlyAtRoot (test_spawn.TestSpawnMultipleSelf) ... [fv-az741-240:130284] PMIX ERROR: NOT_AVAILABLE in file pmix_shmem.c at line 124
[fv-az741-240:130284] PMIX ERROR: ERROR in file gds_shmem.c at line 730
[fv-az741-240:130284] PMIX ERROR: ERROR in file gds_shmem.c at line 802
[fv-az741-240:130284] PMIX ERROR: ERROR in file gds_shmem.c at line 1589
[fv-az741-240:130284] PMIX ERROR: ERROR in file gds_shmem.c at line 1829
[fv-az741-240:130284] PMIX ERROR: UNPACK-FAILURE in file gds_shmem.c at line 1848
[fv-az741-240:130284] PMIX ERROR: UNPACK-FAILURE in file client/pmix_client_fence.c at line 206
[fv-az741-240:130283] PMIX ERROR: NOT_AVAILABLE in file pmix_shmem.c at line 124
[fv-az741-240:130283] PMIX ERROR: ERROR in file gds_shmem.c at line 730
[fv-az741-240:130283] PMIX ERROR: ERROR in file gds_shmem.c at line 802
[fv-az741-240:130283] PMIX ERROR: ERROR in file gds_shmem.c at line 1589
[fv-az741-240:130283] PMIX ERROR: ERROR in file gds_shmem.c at line 1829
[fv-az741-240:130283] PMIX ERROR: UNPACK-FAILURE in file gds_shmem.c at line 1848
[fv-az741-240:130283] PMIX ERROR: UNPACK-FAILURE in file client/pmix_client_fence.c at line 206
[fv-az741-240:130282] PMIX ERROR: NOT_AVAILABLE in file pmix_shmem.c at line 124
[fv-az741-240:130282] PMIX ERROR: ERROR in file gds_shmem.c at line 730
[fv-az741-240:130282] PMIX ERROR: ERROR in file gds_shmem.c at line 802
[fv-az741-240:130282] PMIX ERROR: ERROR in file gds_shmem.c at line 1589
[fv-az741-240:130282] PMIX ERROR: ERROR in file gds_shmem.c at line 1829
[fv-az741-240:130282] PMIX ERROR: UNPACK-FAILURE in file gds_shmem.c at line 1848
[fv-az741-240:130282] PMIX ERROR: UNPACK-FAILURE in file client/pmix_client_fence.c at line 206
[fv-az741-240:130284] OPAL ERROR: File write failure in file dpm/dpm.c at line 652
Traceback (most recent call last):
  File "/home/runner/work/mpi4py-testing/mpi4py-testing/mpi4py/test/spawn_child.py", line 7, in <module>
    parent.Disconnect()
  File "mpi4py/MPI/Comm.pyx", line 1619, in mpi4py.MPI.Comm.Disconnect
mpi4py.MPI.Exception: MPI_ERR_COMM: invalid communicator
```

**Comment by samuelkgutierrez** [comment]
That problem should be addressed by https://github.com/openpmix/openpmix/commit/3811907d905936aa3424841cc1ef4518e6664c86. We enabled by default a new shared-memory component, but found that it still needed work.

**Comment by rhc54** [comment]
Should resolve once https://github.com/open-mpi/ompi/pull/11379 gets committed.

**Comment by dalcinl** [comment]
Fixed via #11379.
